<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Estimate (alpha) diversity measures — estimateDiversity • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Estimate (alpha) diversity measures — estimateDiversity"><meta property="og:description" content="Several functions for calculating (alpha) diversity indices, including
the vegan package options and some others."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.22</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Estimate (alpha) diversity measures</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/HEAD/R/estimateDiversity.R" class="external-link"><code>R/estimateDiversity.R</code></a></small>
    <div class="hidden name"><code>estimateDiversity.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Several functions for calculating (alpha) diversity indices, including
the <code>vegan</code> package options and some others.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">estimateDiversity</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coverage"</span>, <span class="st">"fisher"</span>, <span class="st">"gini_simpson"</span>, <span class="st">"inverse_simpson"</span>,</span>
<span>    <span class="st">"log_modulo_skewness"</span>, <span class="st">"shannon"</span><span class="op">)</span>,</span>
<span>  name <span class="op">=</span> <span class="va">index</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for SummarizedExperiment</span></span>
<span><span class="fu">estimateDiversity</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coverage"</span>, <span class="st">"fisher"</span>, <span class="st">"gini_simpson"</span>, <span class="st">"inverse_simpson"</span>,</span>
<span>    <span class="st">"log_modulo_skewness"</span>, <span class="st">"shannon"</span><span class="op">)</span>,</span>
<span>  name <span class="op">=</span> <span class="va">index</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">SerialParam</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for TreeSummarizedExperiment</span></span>
<span><span class="fu">estimateDiversity</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"coverage"</span>, <span class="st">"faith"</span>, <span class="st">"fisher"</span>, <span class="st">"gini_simpson"</span>, <span class="st">"inverse_simpson"</span>,</span>
<span>    <span class="st">"log_modulo_skewness"</span>, <span class="st">"shannon"</span><span class="op">)</span>,</span>
<span>  name <span class="op">=</span> <span class="va">index</span>,</span>
<span>  tree_name <span class="op">=</span> <span class="st">"phylo"</span>,</span>
<span>  <span class="va">...</span>,</span>
<span>  BPPARAM <span class="op">=</span> <span class="fu">SerialParam</span><span class="op">(</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">estimateFaith</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  tree <span class="op">=</span> <span class="st">"missing"</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"faith"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for SummarizedExperiment,phylo</span></span>
<span><span class="fu">estimateFaith</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  <span class="va">tree</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"faith"</span>,</span>
<span>  node_lab <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for TreeSummarizedExperiment,missing</span></span>
<span><span class="fu">estimateFaith</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"faith"</span>,</span>
<span>  tree_name <span class="op">=</span> <span class="st">"phylo"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>x</dt>
<dd><p>a <code>SummarizedExperiment</code> object or <code>TreeSummarizedExperiment</code>.
The latter is recommended for microbiome data sets and tree-based alpha diversity indices.</p></dd>


<dt>assay.type</dt>
<dd><p>the name of the assay used for
calculation of the sample-wise estimates.</p></dd>


<dt>assay_name</dt>
<dd><p>a single <code>character</code> value for specifying which
assay to use for calculation.
(Please use <code>assay.type</code> instead. At some point <code>assay_name</code>
will be disabled.)</p></dd>


<dt>index</dt>
<dd><p>a <code>character</code> vector, specifying the diversity measures
to be calculated.</p></dd>


<dt>name</dt>
<dd><p>a name for the column(s) of the colData the results should be
stored in. By default this will use the original names of the calculated
indices.</p></dd>


<dt>...</dt>
<dd><p>optional arguments:</p><ul><li><p>threshold: A numeric value in the unit interval,
determining the threshold for coverage index. By default,
<code>threshold</code> is 0.9.</p></li>
<li><p>quantile: Arithmetic abundance classes are evenly cut up to to
this quantile of the data. The assumption is that abundances higher than
this are not common, and they are classified in their own group.
By default, <code>quantile</code> is 0.5.</p></li>
<li><p>num_of_classes: The number of arithmetic abundance classes
from zero to the quantile cutoff indicated by <code>quantile</code>.
By default, <code>num_of_classes</code> is 50.</p></li>
<li><p>only.tips: A boolean value specifying whether to remove internal
nodes when Faith's index is calculated. When <code>only.tips=TRUE</code>, those
rows that are not tips of tree are removed.
(By default: <code>only.tips=FALSE</code>)</p></li>
</ul></dd>


<dt>BPPARAM</dt>
<dd><p>A
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html" class="external-link">BiocParallelParam</a></code>
object specifying whether calculation of estimates should be parallelized.</p></dd>


<dt>tree_name</dt>
<dd><p>a single <code>character</code> value for specifying which
rowTree will be used to calculate faith index.
(By default: <code>tree_name = "phylo"</code>)</p></dd>


<dt>tree</dt>
<dd><p>A phylogenetic tree that is used to calculate 'faith' index.
If <code>x</code> is a <code>TreeSummarizedExperiment</code>, <code>rowTree(x)</code> is
used by default.</p></dd>


<dt>node_lab</dt>
<dd><p>NULL or a character vector specifying the links between rows and
node labels of <code>tree</code>. If a certain row is not linked with the tree, missing
instance should be noted as NA. When NULL, all the rownames should be found from
the tree. (By default: <code>node_lab = NULL</code>)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p><code>x</code> with additional <code>colData</code> named <code>*name*</code></p>


    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The available indices include the ‘Coverage’,
‘Faith's phylogenetic diversity’, ‘Fisher alpha’,
‘Gini-Simpson’,
‘Inverse Simpson’, ‘log-modulo skewness’, and ‘Shannon’
indices. See details for more information and references.</p>
<p>Alpha diversity is a joint quantity that combines elements or community richness
and evenness. Diversity increases, in general, when species richness or
evenness increase.</p>
<p>By default, this function returns all indices.</p>
<ul><li><p>'coverage': Number of species needed to cover a given fraction of
the ecosystem (50 percent by default). Tune this with the threshold
argument.</p></li>
<li><p>'faith': Faith's phylogenetic alpha diversity index measures how
long the taxonomic distance is between taxa that are present in the sample.
Larger values represent higher diversity. Using this index requires
rowTree. (Faith 1992)</p>
<p>If the data includes features that are not in tree's tips but in
internal nodes, there are two options. First, you can keep those features,
and prune the tree to match features so that each tip can be found from
the features. Other option is to remove all features that are not tips.
(See <code>only.tips</code> parameter)</p></li>
<li><p>'fisher': Fisher's alpha; as implemented in
<code><a href="https://rdrr.io/pkg/vegan/man/diversity.html" class="external-link">vegan::fisher.alpha</a></code>. (Fisher et al. 1943)</p></li>
<li><p>'gini_simpson': Gini-Simpson diversity i.e. \(1 - lambda\),
where \(lambda\) is the
Simpson index, calculated as the sum of squared relative abundances.
This corresponds to the diversity index
'simpson' in <code><a href="https://rdrr.io/pkg/vegan/man/diversity.html" class="external-link">vegan::diversity</a></code>.
This is also called Gibbs–Martin, or Blau index in sociology,
psychology and management studies. The Gini-Simpson index (1-lambda)
should not be
confused with Simpson's dominance (lambda), Gini index, or
inverse Simpson index (1/lambda).</p></li>
<li><p>'inverse_simpson': Inverse Simpson diversity:
\(1/lambda\) where \(lambda=sum(p^2)\) and p refers to relative
abundances.
This corresponds to the diversity index
'invsimpson' in vegan::diversity. Don't confuse this with the
closely related Gini-Simpson index</p></li>
<li><p>'log_modulo_skewness': The rarity index characterizes the
concentration of species at low abundance. Here, we use the skewness of
the frequency
distribution of arithmetic abundance classes (see Magurran &amp; McGill 2011).
These are typically right-skewed; to avoid taking log of occasional
negative skews, we follow Locey &amp; Lennon (2016) and use the log-modulo
transformation that adds a value of one to each measure of skewness to
allow logarithmization.</p></li>
<li><p>'shannon': Shannon diversity (entropy).</p></li>
</ul></div>
    <div id="references">
    <h2>References</h2>
    <p>Beisel J-N. et al. (2003)
A Comparative Analysis of Diversity Index Sensitivity.
<em>Internal Rev. Hydrobiol.</em> 88(1):3-15.
<a href="https://portais.ufg.br/up/202/o/2003-comparative_evennes_index.pdf" class="external-link">https://portais.ufg.br/up/202/o/2003-comparative_evennes_index.pdf</a></p>
<p>Bulla L. (1994)
An  index of diversity and its associated diversity measure.
<em>Oikos</em> 70:167--171</p>
<p>Faith D.P. (1992)
Conservation evaluation and phylogenetic diversity.
<em>Biological Conservation</em> 61(1):1-10.</p>
<p>Fisher R.A., Corbet, A.S. &amp; Williams, C.B. (1943)
The relation between the number of species and the number of individuals in
a random sample of animal population.
<em>Journal of Animal Ecology</em> <em>12</em>, 42-58.</p>
<p>Locey K.J. &amp; Lennon J.T. (2016)
Scaling laws predict global microbial diversity.
<em>PNAS</em> 113(21):5970-5975.</p>
<p>Magurran A.E., McGill BJ, eds (2011)
Biological Diversity: Frontiers in Measurement and Assessment.
(Oxford Univ Press, Oxford), Vol 12.</p>
<p>Smith B. &amp; Wilson JB. (1996)
A Consumer's Guide to Diversity Indices.
<em>Oikos</em> 76(1):70-82.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></code></p><ul><li><p><code><a href="estimateRichness.html">estimateRichness</a></code></p></li>
<li><p><code><a href="estimateEvenness.html">estimateEvenness</a></code></p></li>
<li><p><code><a href="estimateDominance.html">estimateDominance</a></code></p></li>
<li><p><code><a href="https://rdrr.io/pkg/vegan/man/diversity.html" class="external-link">diversity</a></code></p></li>
<li><p><code><a href="https://rdrr.io/pkg/vegan/man/specpool.html" class="external-link">estimateR</a></code></p></li>
</ul></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Leo Lahti and Tuomas Borman. Contact: <a href="microbiome.github.io">microbiome.github.io</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># All index names as known by the function</span></span></span>
<span class="r-in"><span><span class="va">index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"shannon"</span>,<span class="st">"gini_simpson"</span>,<span class="st">"inverse_simpson"</span>, <span class="st">"coverage"</span>, <span class="st">"fisher"</span>, </span></span>
<span class="r-in"><span><span class="st">"faith"</span>,  <span class="st">"log_modulo_skewness"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Corresponding polished names</span></span></span>
<span class="r-in"><span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Shannon"</span>,<span class="st">"GiniSimpson"</span>,<span class="st">"InverseSimpson"</span>, <span class="st">"Coverage"</span>, <span class="st">"Fisher"</span>, </span></span>
<span class="r-in"><span><span class="st">"Faith"</span>,  <span class="st">"LogModSkewness"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Calculate diversities</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">estimateDiversity</span><span class="op">(</span><span class="va">tse</span>, index <span class="op">=</span> <span class="va">index</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># The colData contains the indices with their code names by default</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span>, <span class="va">index</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 26 rows and 7 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           shannon gini_simpson inverse_simpson  coverage    fisher     faith</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;numeric&gt;    &lt;numeric&gt;       &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3       6.57652     0.994656        187.1298      1113  1035.008   250.535</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1       6.77660     0.995212        208.8436      1374  1107.495   262.263</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1       6.49849     0.996290        269.5441       829   854.279   208.458</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw   3.82837     0.927599         13.8119        71   313.719   117.876</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw   3.28767     0.909738         11.0789        37   289.979   119.825</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...           ...          ...             ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS28      4.12654     0.965175         28.7152        62   337.892   124.651</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS29      3.45277     0.918098         12.2097        46   318.969   121.206</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even1     4.08366     0.968198         31.4447        51   546.578   179.938</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even2     3.95691     0.963916         27.7129        47   401.760   140.810</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even3     4.00637     0.967340         30.6189        49   344.939   126.569</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         log_modulo_skewness</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                   &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3                 2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1                 2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1                 2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw             2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw             2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...                     ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS28                2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS29                2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even1               2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even2               2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even3               2.06141</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Removing indices</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span>, <span class="va">index</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="cn">NULL</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># 'threshold' can be used to determine threshold for 'coverage' index</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">estimateDiversity</span><span class="op">(</span><span class="va">tse</span>, index <span class="op">=</span> <span class="st">"coverage"</span>, threshold <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># 'quantile' and 'num_of_classes' can be used when</span></span></span>
<span class="r-in"><span><span class="co"># 'log_modulo_skewness' is calculated</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">estimateDiversity</span><span class="op">(</span><span class="va">tse</span>, index <span class="op">=</span> <span class="st">"log_modulo_skewness"</span>,</span></span>
<span class="r-in"><span>       quantile <span class="op">=</span> <span class="fl">0.75</span>, num_of_classes <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># It is recommended to specify also the final names used in the output.</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">estimateDiversity</span><span class="op">(</span><span class="va">tse</span>,</span></span>
<span class="r-in"><span>  index <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"shannon"</span>, <span class="st">"gini_simpson"</span>, <span class="st">"inverse_simpson"</span>, <span class="st">"coverage"</span>,</span></span>
<span class="r-in"><span>               <span class="st">"fisher"</span>, <span class="st">"faith"</span>, <span class="st">"log_modulo_skewness"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   name <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Shannon"</span>, <span class="st">"GiniSimpson"</span>,  <span class="st">"InverseSimpson"</span>,  <span class="st">"Coverage"</span>,</span></span>
<span class="r-in"><span>               <span class="st">"Fisher"</span>, <span class="st">"Faith"</span>, <span class="st">"LogModSkewness"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># The colData contains the indices by their new names provided by the user</span></span></span>
<span class="r-in"><span><span class="fu">colData</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span>, <span class="va">name</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> DataFrame with 26 rows and 7 columns</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           Shannon GiniSimpson InverseSimpson  Coverage    Fisher     Faith</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         &lt;numeric&gt;   &lt;numeric&gt;      &lt;numeric&gt; &lt;integer&gt; &lt;numeric&gt; &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3       6.57652    0.994656       187.1298      1113  1035.008   250.535</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1       6.77660    0.995212       208.8436      1374  1107.495   262.263</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1       6.49849    0.996290       269.5441       829   854.279   208.458</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw   3.82837    0.927599        13.8119        71   313.719   117.876</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw   3.28767    0.909738        11.0789        37   289.979   119.825</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...           ...         ...            ...       ...       ...       ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS28      4.12654    0.965175        28.7152        62   337.892   124.651</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS29      3.45277    0.918098        12.2097        46   318.969   121.206</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even1     4.08366    0.968198        31.4447        51   546.578   179.938</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even2     3.95691    0.963916        27.7129        47   401.760   140.810</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even3     4.00637    0.967340        30.6189        49   344.939   126.569</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         LogModSkewness</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              &lt;numeric&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3            2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1            2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1            2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw        2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw        2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ...                ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS28           2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> TS29           2.06142</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even1          2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even2          2.06141</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Even3          2.06141</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Compare the indices visually</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/graphics/pairs.html" class="external-link">pairs</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span>, <span class="va">name</span><span class="op">]</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="estimateDiversity-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Plotting the diversities - use the selected names</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"Shannon"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="estimateDiversity-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># ... by sample type</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/scater/man/plotColData.html" class="external-link">plotColData</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"Shannon"</span>, <span class="st">"SampleType"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="estimateDiversity-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co"># combining different plots</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">plot_index</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Shannon"</span>,<span class="st">"GiniSimpson"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">plot_index</span>,</span></span>
<span class="r-in"><span>               <span class="va">plotColData</span>,</span></span>
<span class="r-in"><span>               object <span class="op">=</span> <span class="va">tse</span>,</span></span>
<span class="r-in"><span>               x <span class="op">=</span> <span class="st">"SampleType"</span>,</span></span>
<span class="r-in"><span>               colour_by <span class="op">=</span> <span class="st">"SampleType"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">plots</span>,<span class="st">"+"</span>,</span></span>
<span class="r-in"><span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle<span class="op">=</span><span class="fl">45</span>,hjust<span class="op">=</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">plots</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">plot_index</span></span></span>
<span class="r-in"><span><span class="va">plots</span><span class="op">$</span><span class="va">Shannon</span> <span class="op">+</span> <span class="va">plots</span><span class="op">$</span><span class="va">GiniSimpson</span> <span class="op">+</span> <span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>guides <span class="op">=</span> <span class="st">"collect"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="estimateDiversity-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

