<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculate dissimilarities — addDissimilarity • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate dissimilarities — addDissimilarity"><meta property="og:description" content="These functions are designed to calculate dissimilarities on data stored
within a
TreeSummarizedExperiment
object. For overlap, Unifrac, and Jensen-Shannon Divergence (JSD)
dissimilarities, the functions use mia internal functions, while for other
types of dissimilarities, they rely on vegdist
by default."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.23</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculate dissimilarities</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/master/R/AllGenerics.R" class="external-link"><code>R/AllGenerics.R</code></a>, <a href="https://github.com/microbiome/mia/blob/master/R/addDissimilarity.R" class="external-link"><code>R/addDissimilarity.R</code></a></small>
    <div class="hidden name"><code>getDissimilarity.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions are designed to calculate dissimilarities on data stored
within a
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
object. For overlap, Unifrac, and Jensen-Shannon Divergence (JSD)
dissimilarities, the functions use mia internal functions, while for other
types of dissimilarities, they rely on <code><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></code>
by default.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">addDissimilarity</span><span class="op">(</span><span class="va">x</span>, <span class="va">method</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getDissimilarity</span><span class="op">(</span><span class="va">x</span>, <span class="va">method</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">addDissimilarity</span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"bray"</span>, name <span class="op">=</span> <span class="va">method</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">getDissimilarity</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  niter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  transposed <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">getDissimilarity</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  niter <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  transposed <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ANY'</span></span>
<span><span class="fu">getDissimilarity</span><span class="op">(</span><span class="va">x</span>, method <span class="op">=</span> <span class="st">"bray"</span>, niter <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p><code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>
or <code>matrix</code>.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p><code>Character scalar</code>. Specifies which dissimilarity to
calculate. (Default: <code>"bray"</code>)</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>other arguments passed into <code><a href="https://rdrr.io/pkg/vegan/man/avgdist.html" class="external-link">avgdist</a></code>,
<code><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></code>, or into mia internal functions:</p>
<ul><li><p><code>sample</code>: The sampling depth in rarefaction.
(Default: <code>min(rowSums2(x))</code>)</p></li>
<li><p><code>dis.fun</code>: <code>Character scalar</code>. Specifies the dissimilarity
function to be used.</p></li>
<li><p><code>transf</code>: <code>Function</code>. Specifies the optional
transformation applied before calculating the dissimilarity matrix.</p></li>
<li><p><code>tree.name</code>: (Unifrac)  <code>Character scalar</code>. Specifies the
name of the tree from <code>rowTree(x)</code> that is used in calculation.
Disabled when <code>tree</code> is specified. (Default: <code>"phylo"</code>)</p></li>
<li><p><code>tree</code>: (Unifrac) <code>phylo</code>. A phylogenetic tree used in
calculation. (Default: <code>NULL</code>)</p></li>
<li><p><code>weighted</code>: (Unifrac) <code>Logical scalar</code>. Should use
weighted-Unifrac calculation?
Weighted-Unifrac takes into account the relative abundance of
species/taxa shared between samples, whereas unweighted-Unifrac only
considers presence/absence. Default is <code>FALSE</code>, meaning the
unweighted-Unifrac dissimilarity is calculated for all pairs of samples.
(Default: <code>FALSE</code>)</p></li>
<li><p><code>node.label</code> (Unifrac) <code>character vector</code>. Used only if
<code>x</code> is a matrix. Specifies links between rows/columns and tips of
<code>tree</code>. The length must equal the number of rows/columns of <code>x</code>.
Furthermore, all the node labs must be present in <code>tree</code>.</p></li>
<li><p><code>chunkSize</code>: (JSD) <code>Integer scalar</code>. Defines the size of
data  send to the individual worker. Only has an effect, if <code>BPPARAM</code>
defines more than one worker. (Default: <code>nrow(x)</code>)</p></li>
<li><p><code>BPPARAM</code>: (JSD)
<code><a href="https://rdrr.io/pkg/BiocParallel/man/BiocParallelParam-class.html" class="external-link">BiocParallelParam</a></code>.
Specifies whether the calculation should be parallelized.</p></li>
<li><p><code>detection</code>: (Overlap) <code>Numeric scalar</code>.
Defines detection threshold for absence/presence of features. Feature that
has abundance under threshold in either of samples, will be discarded when
evaluating overlap between samples. (Default: <code>0</code>)</p></li>
</ul></dd>


<dt id="arg-name">name<a class="anchor" aria-label="anchor" href="#arg-name"></a></dt>
<dd><p><code>Character scalar</code>. The name to be used to store the result
in metadata of the output. (Default: <code>method</code>)</p></dd>


<dt id="arg-assay-type">assay.type<a class="anchor" aria-label="anchor" href="#arg-assay-type"></a></dt>
<dd><p><code>Character scalar</code>. Specifies the name of assay
used in calculation. (Default: <code>"counts"</code>)</p></dd>


<dt id="arg-niter">niter<a class="anchor" aria-label="anchor" href="#arg-niter"></a></dt>
<dd><p><code>Integer scalar</code>. Specifies the number of
rarefaction rounds. Rarefaction is not applied when <code>niter=NULL</code>
(see Details section). (Default: <code>NULL</code>)</p></dd>


<dt id="arg-transposed">transposed<a class="anchor" aria-label="anchor" href="#arg-transposed"></a></dt>
<dd><p><code>Logical scalar</code>. Specifies if x is transposed with
cells in rows. (Default: <code>FALSE</code>)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>getDissimilarity</code> returns a sample-by-sample dissimilarity matrix.</p>
<p><code>addDissimilarity</code> returns <code>x</code> that includes dissimilarity matrix
in its metadata.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Overlap reflects similarity between sample-pairs. When overlap is
calculated using relative abundances, the higher the value the higher the
similarity is. When using relative abundances, overlap value 1 means that
all the abundances of features are equal between two samples, and 0 means
that samples have completely different relative abundances.</p>
<p>Unifrac is calculated with <code><a href="https://cmmr.github.io/rbiom/reference/rbiom-deprecated.html" class="external-link">rbiom:unifrac()</a></code>.</p>
<p>If rarefaction is enabled, <code><a href="https://rdrr.io/pkg/vegan/man/avgdist.html" class="external-link">vegan:avgdist()</a></code> is
utilized.</p>
<p>Rarefaction can be used to control uneven sequencing depths. Although,
it is highly debated method. Some think that it is the only option that
successfully controls the variation caused by uneven sampling depths.
The biggest argument against rarefaction is the fact that it omits data.</p>
<p>Rarefaction works by sampling the counts randomly. This random sampling
is done <code>niter</code> times. In each sampling iteration, <code>sample</code> number
of random samples are drawn, and dissimilarity is calculated for this
subset. After the iterative process, there are <code>niter</code> number of
result that are then averaged to get the final result.</p>
<p>Refer to Schloss (2024) for more details on rarefaction.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>For unifrac dissimilarity: <a href="http://bmf.colorado.edu/unifrac/" class="external-link">http://bmf.colorado.edu/unifrac/</a></p>
<p>See also additional descriptions of Unifrac in the following articles:</p>
<p>Lozupone, Hamady and Knight, “Unifrac - An Online Tool for Comparing
Microbial Community Diversity in a Phylogenetic Context.”, BMC
Bioinformatics 2006, 7:371</p>
<p>Lozupone, Hamady, Kelley and Knight, “Quantitative and qualitative (beta)
diversity measures lead to different insights into factors that structure
microbial communities.” Appl Environ Microbiol. 2007</p>
<p>Lozupone C, Knight R. “Unifrac: a new phylogenetic method for comparing
microbial communities.” Appl Environ Microbiol. 2005 71 (12):8228-35.</p>
<p>For JSD dissimilarity:
Jensen-Shannon Divergence and Hilbert space embedding.
Bent Fuglede and Flemming Topsoe University of Copenhagen,
Department of Mathematics
<a href="http://www.math.ku.dk/~topsoe/ISIT2004JSD.pdf" class="external-link">http://www.math.ku.dk/~topsoe/ISIT2004JSD.pdf</a></p>
<p>For rarefaction:
Schloss PD (2024) Rarefaction is currently the best approach to control for
uneven sequencing effort in amplicon sequence analyses. <em>mSphere</em>
28;9(2):e0035423. doi: 10.1128/msphere.00354-23</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><a href="http://en.wikipedia.org/wiki/Jensen-Shannon_divergence" class="external-link">http://en.wikipedia.org/wiki/Jensen-Shannon_divergence</a></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/microbiome/mia" class="external-link">mia</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load dataset</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Overlap dissimilarity</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">addDissimilarity</span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"overlap"</span>, detection <span class="op">=</span> <span class="fl">0.25</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="st">"overlap"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               CL3       CC1     SV1 M31Fcsw M11Fcsw M31Plmr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3           0.0  985499.5  692281  951745 1217896  466124</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1      985499.5       0.0  848128  975408 1253552  626083</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1      692281.0  848128.0       0  808197 1083470  493475</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw  951745.0  975408.0  808197       0 1785250 1014236</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw 1217896.5 1253551.5 1083470 1785250       0 1281626</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Plmr  466124.0  626083.0  493475 1014236 1281626       0</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### JSD dissimilarity</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">addDissimilarity</span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"jsd"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="st">"jsd"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               CL3       CC1       SV1   M31Fcsw   M11Fcsw   M31Plmr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3     0.0000000 0.2547080 0.5051481 0.6809263 0.6813418 0.6713074</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1     0.2547080 0.0000000 0.4378036 0.6863213 0.6869444 0.6620168</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1     0.5051481 0.4378036 0.0000000 0.6848186 0.6850675 0.6649845</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw 0.6809263 0.6863213 0.6848186 0.0000000 0.2864774 0.6720375</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw 0.6813418 0.6869444 0.6850675 0.2864774 0.0000000 0.6831013</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Plmr 0.6713074 0.6620168 0.6649845 0.6720375 0.6831013 0.0000000</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Multi Dimensional Scaling applied to JSD dissimilarity matrix</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="addMDS.html">addMDS</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"overlap"</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/reducedDims.html" class="external-link">reducedDim</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"MDS"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>                [,1]      [,2]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3      -43284.813 -45143.87</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1       -4790.904 -40203.23</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1       26147.150 -15952.00</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw -133864.366 -14824.29</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw  610755.430 803373.51</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Plmr  -78217.650   3434.70</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Unifrac dissimilarity</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">getDissimilarity</span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"unifrac"</span>, weighted <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 26 26</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">addDissimilarity</span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"unifrac"</span>, weighted <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">metadata</span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[[</span><span class="st">"unifrac"</span><span class="op">]</span><span class="op">]</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               CL3       CC1       SV1   M31Fcsw   M11Fcsw   M31Plmr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3     0.0000000 0.1720386 0.2817016 0.5772531 0.5625487 0.4504863</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1     0.1720386 0.0000000 0.2473207 0.5782640 0.5632083 0.4493349</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1     0.2817016 0.2473207 0.0000000 0.5701039 0.5517097 0.4609602</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw 0.5772531 0.5782640 0.5701039 0.0000000 0.1924949 0.5244770</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw 0.5625487 0.5632083 0.5517097 0.1924949 0.0000000 0.5484897</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Plmr 0.4504863 0.4493349 0.4609602 0.5244770 0.5484897 0.0000000</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">### Bray dissimilarity</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Bray is usually applied to relative abundances so we have to apply</span></span></span>
<span class="r-in"><span><span class="co"># transformation first</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">getDissimilarity</span><span class="op">(</span><span class="va">tse</span>, method <span class="op">=</span> <span class="st">"bray"</span>, assay.type <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               CL3       CC1       SV1   M31Fcsw   M11Fcsw   M31Plmr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3     0.0000000 0.5902949 0.8438458 0.9947506 0.9948792 0.9844500</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1     0.5902949 0.0000000 0.7835588 0.9969718 0.9972773 0.9765050</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1     0.8438458 0.7835588 0.0000000 0.9961894 0.9963561 0.9815678</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw 0.9947506 0.9969718 0.9961894 0.0000000 0.5468973 0.9874252</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw 0.9948792 0.9972773 0.9963561 0.5468973 0.0000000 0.9955337</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Plmr 0.9844500 0.9765050 0.9815678 0.9874252 0.9955337 0.0000000</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># If applying rarefaction, the input must be count matrix and transformation</span></span></span>
<span class="r-in"><span><span class="co"># method specified in function call (Note: increase niter)</span></span></span>
<span class="r-in"><span><span class="va">rclr</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span></span></span>
<span class="r-in"><span>    <span class="fu">vegan</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/decostand.html" class="external-link">decostand</a></span><span class="op">(</span><span class="va">x</span>, method<span class="op">=</span><span class="st">"rclr"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu">getDissimilarity</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">tse</span>, method <span class="op">=</span> <span class="st">"euclidean"</span>, transf <span class="op">=</span> <span class="va">rclr</span>, niter <span class="op">=</span> <span class="fl">2L</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               CL3       CC1       SV1   M31Fcsw  M11Fcsw   M31Plmr</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CL3       0.00000  92.41533 111.91988 100.26503 96.37790 104.02765</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> CC1      92.41533   0.00000 107.63145 101.04726 96.98399 105.23199</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> SV1     111.91988 107.63145   0.00000  97.99801 94.11056 102.95526</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Fcsw 100.26503 101.04726  97.99801   0.00000 53.73290  80.20711</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M11Fcsw  96.37790  96.98399  94.11056  53.73290  0.00000  75.98772</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> M31Plmr 104.02765 105.23199 102.95526  80.20711 75.98772   0.00000</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Tuomas Borman, Felix G.M. Ernst, Sudarshan A. Shetty, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

