<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Calculation prevalence information for features across samples — getPrevalence • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Calculation prevalence information for features across samples — getPrevalence"><meta property="og:description" content="These functions calculate the population prevalence for taxonomic ranks in a
SummarizedExperiment object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.23</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Calculation prevalence information for features across samples</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/master/R/AllGenerics.R" class="external-link"><code>R/AllGenerics.R</code></a>, <a href="https://github.com/microbiome/mia/blob/master/R/getPrevalence.R" class="external-link"><code>R/getPrevalence.R</code></a></small>
    <div class="hidden name"><code>getPrevalence.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>These functions calculate the population prevalence for taxonomic ranks in a
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment</a></code> object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">getPrevalence</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getPrevalent</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getRare</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">subsetByPrevalent</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">subsetByRare</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">getPrevalentAbundance</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="va">assay_name</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">addPrevalence</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">addPrevalence</span><span class="op">(</span><span class="va">x</span>, name <span class="op">=</span> <span class="st">"prevalence"</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ANY'</span></span>
<span><span class="fu">getPrevalence</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  detection <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  include.lowest <span class="op">=</span> <span class="va">include_lowest</span>,</span>
<span>  include_lowest <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  sort <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  na.rm <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">getPrevalence</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="va">assay_name</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  rank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ANY'</span></span>
<span><span class="fu">getPrevalent</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span>,</span>
<span>  include.lowest <span class="op">=</span> <span class="va">include_lowest</span>,</span>
<span>  include_lowest <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">getPrevalent</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  rank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span>,</span>
<span>  include.lowest <span class="op">=</span> <span class="va">include_lowest</span>,</span>
<span>  include_lowest <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ANY'</span></span>
<span><span class="fu">getRare</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span>,</span>
<span>  include.lowest <span class="op">=</span> <span class="va">include_lowest</span>,</span>
<span>  include_lowest <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">getRare</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  rank <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span>,</span>
<span>  include.lowest <span class="op">=</span> <span class="va">include_lowest</span>,</span>
<span>  include_lowest <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">subsetByPrevalent</span><span class="op">(</span><span class="va">x</span>, rank <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">subsetByPrevalent</span><span class="op">(</span><span class="va">x</span>, update.tree <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">subsetByRare</span><span class="op">(</span><span class="va">x</span>, rank <span class="op">=</span> <span class="cn">NULL</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'TreeSummarizedExperiment'</span></span>
<span><span class="fu">subsetByRare</span><span class="op">(</span><span class="va">x</span>, update.tree <span class="op">=</span> <span class="cn">TRUE</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'ANY'</span></span>
<span><span class="fu">getPrevalentAbundance</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="va">assay_name</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"relabundance"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">getPrevalentAbundance</span><span class="op">(</span><span class="va">x</span>, assay.type <span class="op">=</span> <span class="va">assay_name</span>, assay_name <span class="op">=</span> <span class="st">"counts"</span>, <span class="va">...</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p><code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>additional arguments</p><ul><li><p>If <code>!is.null(rank)</code> arguments are passed on to
<code><a href="agglomerate-methods.html">agglomerateByRank</a></code>. See
<code><a href="agglomerate-methods.html">?agglomerateByRank</a></code> for more details.</p></li>
<li><p>for <code>getPrevalent</code>, <code>getRare</code>, <code>subsetByPrevalent</code>
and <code>subsetByRare</code> additional parameters passed to
<code>getPrevalence</code></p></li>
<li><p>for <code>getPrevalentAbundance</code> additional parameters passed to
<code>getPrevalent</code></p></li>
</ul></dd>


<dt id="arg-assay-type">assay.type<a class="anchor" aria-label="anchor" href="#arg-assay-type"></a></dt>
<dd><p><code>Character scalar</code>. Specifies the name of assay
used in calculation. (Default: <code>"counts"</code>)</p></dd>


<dt id="arg-assay-name">assay_name<a class="anchor" aria-label="anchor" href="#arg-assay-name"></a></dt>
<dd><p>Deprecated. Use <code>assay.type</code> instead.</p></dd>


<dt id="arg-name">name<a class="anchor" aria-label="anchor" href="#arg-name"></a></dt>
<dd><p><code>Character scalar</code>. Specifies name of column in
<code>rowData</code> where the results will be stored.
(Default: <code>"prevalence"</code>)</p></dd>


<dt id="arg-detection">detection<a class="anchor" aria-label="anchor" href="#arg-detection"></a></dt>
<dd><p><code>Numeric scalar</code>. Detection threshold for
absence/presence. If <code>as_relative = FALSE</code>,
it sets the counts threshold for a taxon to be considered present.
If <code>as_relative = TRUE</code>, it sets the relative abundance threshold
for a taxon to be considered present. (Default: <code>0</code>)</p></dd>


<dt id="arg-include-lowest">include.lowest<a class="anchor" aria-label="anchor" href="#arg-include-lowest"></a></dt>
<dd><p><code>Logical scalar</code>. Should the lower boundary of the
detection and prevalence cutoffs be included? (Default: <code>FALSE</code>)</p></dd>


<dt id="arg-include-lowest">include_lowest<a class="anchor" aria-label="anchor" href="#arg-include-lowest"></a></dt>
<dd><p>Deprecated. Use <code>include.lowest</code> instead.</p></dd>


<dt id="arg-sort">sort<a class="anchor" aria-label="anchor" href="#arg-sort"></a></dt>
<dd><p><code>Logical scalar</code>. Should the result be sorted by prevalence?
(Default: <code>FALSE</code>)</p></dd>


<dt id="arg-na-rm">na.rm<a class="anchor" aria-label="anchor" href="#arg-na-rm"></a></dt>
<dd><p><code>Logical scalar</code>. Should NA values be omitted?
(Default: <code>TRUE</code>)</p></dd>


<dt id="arg-rank">rank<a class="anchor" aria-label="anchor" href="#arg-rank"></a></dt>
<dd><p><code>Character scalar</code>. Defines a taxonomic rank. Must be a
value of <code><a href="taxonomy-methods.html">taxonomyRanks()</a></code> function.</p></dd>


<dt id="arg-prevalence">prevalence<a class="anchor" aria-label="anchor" href="#arg-prevalence"></a></dt>
<dd><p>Prevalence threshold (in 0 to 1). The
required prevalence is strictly greater by default. To include the
limit, set <code>include.lowest</code> to <code>TRUE</code>.</p></dd>


<dt id="arg-update-tree">update.tree<a class="anchor" aria-label="anchor" href="#arg-update-tree"></a></dt>
<dd><p><code>Logical scalar</code>. Should
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html" class="external-link">rowTree()</a></code> also be agglomerated? (Default: <code>TRUE</code>)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>subsetPrevalent</code> and <code>subsetRareFeatures</code> return subset of
<code>x</code>.</p>
<p>All other functions return a named vectors:</p><ul><li><p><code>getPrevalence</code> returns a <code>numeric</code> vector with the
names being set to either the row names of <code>x</code> or the names after
agglomeration. <code>addPrevalence</code> adds these results to
<code>rowData(x)</code>.</p></li>
<li><p><code>getPrevalentAbundance</code> returns a <code>numeric</code> vector with
the names corresponding to the column name of <code>x</code> and include the
joint abundance of prevalent taxa.</p></li>
<li><p><code>getPrevalent</code> and <code>getRare</code> return a
<code>character</code> vector with only the names exceeding the threshold set
by <code>prevalence</code>, if the <code>rownames</code> of <code>x</code> is set.
Otherwise an <code>integer</code> vector is returned matching the rows in
<code>x</code>.</p></li>
</ul></div>
    <div id="details">
    <h2>Details</h2>
    <p><code>getPrevalence</code> calculates the frequency of samples that exceed
the detection threshold. For <code>SummarizedExperiment</code> objects, the
prevalence is calculated for the selected taxonomic rank, otherwise for the
rows. The absolute population prevalence can be obtained by multiplying the
prevalence by the number of samples (<code>ncol(x)</code>).</p>
<p>The core abundance index from <code>getPrevalentAbundance</code> gives the relative
proportion of the core species (in between 0 and 1). The core taxa are
defined as those that exceed the given population prevalence threshold at the
given detection level as set for <code>getPrevalent</code>.</p>
<p><code>subsetPrevalent</code> and <code>subsetRareFeatures</code> return a subset of
<code>x</code>.
The subset includes the most prevalent or rare taxa that are calculated with
<code>getPrevalent</code> or <code>getRare</code> respectively.</p>
<p><code>getPrevalent</code> returns taxa that are more prevalent with the
given detection threshold for the selected taxonomic rank.</p>
<p><code>getRare</code> returns complement of <code>getPrevalent</code>.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>A Salonen et al. The adult intestinal core microbiota is determined by
analysis depth and health status. Clinical Microbiology and Infection
18(S4):16 20, 2012.
To cite the R package, see citation('mia')</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="agglomerate-methods.html">agglomerateByRank</a></code>,
<code><a href="summary.html">getTop</a></code></p></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span><span class="co"># Get prevalence estimates for individual ASV/OTU</span></span></span>
<span class="r-in"><span><span class="va">prevalence.frequency</span> <span class="op">&lt;-</span> <span class="fu">getPrevalence</span><span class="op">(</span><span class="va">tse</span>,</span></span>
<span class="r-in"><span>                                      detection <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>                                      sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prevalence.frequency</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 145149 114821 108747 526804 332405  98605 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      1      1      1      1      1      1 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Get prevalence estimates for phyla</span></span></span>
<span class="r-in"><span><span class="co"># - the getPrevalence function itself always returns population frequencies</span></span></span>
<span class="r-in"><span><span class="va">prevalence.frequency</span> <span class="op">&lt;-</span> <span class="fu">getPrevalence</span><span class="op">(</span><span class="va">tse</span>,</span></span>
<span class="r-in"><span>                                      rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span></span>
<span class="r-in"><span>                                      detection <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>                                      sort <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prevalence.frequency</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             WS3           WPS-2 Verrucomicrobia     Tenericutes    Spirochaetes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               1               1               1               1               1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Proteobacteria </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               1 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># - to obtain population counts, multiply frequencies with the sample size,</span></span></span>
<span class="r-in"><span><span class="co"># which answers the question "In how many samples is this phylum detectable"</span></span></span>
<span class="r-in"><span><span class="va">prevalence.count</span> <span class="op">&lt;-</span> <span class="va">prevalence.frequency</span> <span class="op">*</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prevalence.count</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             WS3           WPS-2 Verrucomicrobia     Tenericutes    Spirochaetes </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              26              26              26              26              26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>  Proteobacteria </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>              26 </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Detection threshold 1 (strictly greater by default);</span></span></span>
<span class="r-in"><span><span class="co"># Note that the data (GlobalPatterns) is here in absolute counts</span></span></span>
<span class="r-in"><span><span class="co"># (and not compositional, relative abundances)</span></span></span>
<span class="r-in"><span><span class="co"># Prevalence threshold 50 percent (strictly greater by default)</span></span></span>
<span class="r-in"><span><span class="va">prevalent</span> <span class="op">&lt;-</span> <span class="fu">getPrevalent</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">tse</span>,</span></span>
<span class="r-in"><span>    rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span></span>
<span class="r-in"><span>    detection <span class="op">=</span> <span class="fl">10</span>,</span></span>
<span class="r-in"><span>    prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">prevalent</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Acidobacteria"  "Actinobacteria" "Bacteroidetes"  "Chlorobi"      </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5] "Chloroflexi"    "Crenarchaeota" </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Add relative aundance data</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">tse</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span>, method <span class="op">=</span> <span class="st">"relabundance"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Gets a subset of object that includes prevalent taxa</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalent"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">subsetByPrevalent</span><span class="op">(</span><span class="va">tse</span>,</span></span>
<span class="r-in"><span>                                             rank <span class="op">=</span> <span class="st">"Family"</span>,</span></span>
<span class="r-in"><span>                                             assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span></span>
<span class="r-in"><span>                                             detection <span class="op">=</span> <span class="fl">0.001</span>,</span></span>
<span class="r-in"><span>                                             prevalence <span class="op">=</span> <span class="fl">0.55</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"prevalent"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 3 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(1): agglomerated_by_rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(2): counts relabundance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(3): Flavobacteriaceae Lachnospiraceae Rhodobacteraceae_1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (3 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (3 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># getRare returns the inverse</span></span></span>
<span class="r-in"><span><span class="va">rare</span> <span class="op">&lt;-</span> <span class="fu">getRare</span><span class="op">(</span><span class="va">tse</span>,</span></span>
<span class="r-in"><span>    rank <span class="op">=</span> <span class="st">"Phylum"</span>,</span></span>
<span class="r-in"><span>    assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span></span>
<span class="r-in"><span>    detection <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">100</span>,</span></span>
<span class="r-in"><span>    prevalence <span class="op">=</span> <span class="fl">50</span><span class="op">/</span><span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">rare</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "ABY1_OD1"        "AC1"             "AD3"             "Acidobacteria"  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [5] "Armatimonadetes" "BRC1"           </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Gets a subset of object that includes rare taxa</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"rare"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">subsetByRare</span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="va">tse</span>,</span></span>
<span class="r-in"><span>    rank <span class="op">=</span> <span class="st">"Class"</span>,</span></span>
<span class="r-in"><span>    assay.type <span class="op">=</span> <span class="st">"relabundance"</span>,</span></span>
<span class="r-in"><span>    detection <span class="op">=</span> <span class="fl">0.001</span>,</span></span>
<span class="r-in"><span>    prevalence <span class="op">=</span> <span class="fl">0.001</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"rare"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 71 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(1): agglomerated_by_rank</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(2): counts relabundance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(71): 09D2Y74 12-24 ... koll11 vadinHA49</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (71 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (71 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Names of both experiments, prevalent and rare, can be found from slot</span></span></span>
<span class="r-in"><span><span class="co"># altExpNames</span></span></span>
<span class="r-in"><span><span class="va">tse</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 19216 26 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(2): counts relabundance</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(19216): 549322 522457 ... 200359 271582</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(26): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(2): prevalent rare</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (19216 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">getPrevalentAbundance</span><span class="op">(</span><span class="va">esophagus</span>, assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         B         C         D </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0.9605911 0.8980392 0.9086758 </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Tuomas Borman, Felix G.M. Ernst, Sudarshan A. Shetty, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

