<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Import Metaphlan results to TreeSummarizedExperiment — importMetaPhlAn • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Import Metaphlan results to TreeSummarizedExperiment — importMetaPhlAn"><meta property="og:description" content="Import Metaphlan results to TreeSummarizedExperiment"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.22</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Import Metaphlan results to <code>TreeSummarizedExperiment</code></h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/HEAD/R/importMetaphlan.R" class="external-link"><code>R/importMetaphlan.R</code></a></small>
    <div class="hidden name"><code>importMetaPhlAn.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Import Metaphlan results to <code>TreeSummarizedExperiment</code></p>
    </div>


    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>file</dt>
<dd><p>a single <code>character</code> value defining the file
path of the Metaphlan file. The file must be in merged Metaphlan format.</p></dd>


<dt>colData</dt>
<dd><p>a DataFrame-like object that includes sample names in
rownames, or a single <code>character</code> value defining the file
path of the sample metadata file. The file must be in <code>tsv</code> format
(default: <code>colData = NULL</code>).</p></dd>


<dt>sample_meta</dt>
<dd><p>a DataFrame-like object that includes sample names in
rownames, or a single <code>character</code> value defining the file
path of the sample metadata file. The file must be in <code>tsv</code> format
(default: <code>sample_meta = NULL</code>).</p></dd>


<dt>phy_tree</dt>
<dd><p>a single <code>character</code> value defining the file
path of the phylogenetic tree.
(default: <code>phy_tree = NULL</code>).</p></dd>


<dt>...</dt>
<dd><p>additional arguments:</p><ul><li><p><code>assay.type</code>: A single character value for naming
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></code>
(default: <code>assay.type = "counts"</code>)</p></li>
<li><p><code>assay_name</code>: A single <code>character</code> value for specifying which
assay to use for calculation. (Please use <code>assay.type</code> instead.
At some point <code>assay_name</code> will be disabled.)</p></li>
<li><p><code>removeTaxaPrefixes</code>: <code>TRUE</code> or <code>FALSE</code>: Should
taxonomic prefixes be removed? (default:
<code>removeTaxaPrefixes = FALSE</code>)</p></li>
<li><p><code>remove.suffix</code>: <code>TRUE</code> or <code>FALSE</code>: Should
suffixes of sample names be removed? Metaphlan pipeline adds suffixes
to sample names. Suffixes are formed from file names. By selecting
<code>remove.suffix = TRUE</code>, you can remove pattern from end of sample
names that is shared by all. (default: <code>remove.suffix = FALSE</code>)</p></li>
<li><p><code>set.ranks</code>: <code>TRUE</code> or <code>FALSE</code>: Should the columns
in the rowData that are treated as taxonomy ranks be updated according to
the ranks found in the imported data?
(default: <code>set.ranks = FALSE</code>)</p></li>
</ul></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A
<code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code></p>


<p>object</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Import Metaphlan (versions 2, 3 and 4 supported) results.
Input must be in merged Metaphlan format.
(See
<a href="https://github.com/biobakery/MetaPhlAn/wiki/MetaPhlAn-4#merging-tables" class="external-link">
the Metaphlan documentation and <code>merge_metaphlan_tables</code> method.</a>)
Data is imported so that data at the lowest rank is imported as a
<code>TreeSummarizedExperiment</code> object. Data at higher rank is imported as a
<code>SummarizedExperiment</code> objects which are stored to <code>altExp</code> of
<code>TreeSummarizedExperiment</code> object.</p>
<p>Currently Metaphlan versions 2, 3, and 4 are supported.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Beghini F, McIver LJ, Blanco-Míguez A, Dubois L, Asnicar F, Maharjan S, Mailyan A,
Manghi P, Scholz M, Thomas AM, Valles-Colomer M, Weingart G, Zhang Y, Zolfo M,
Huttenhower C, Franzosa EA, &amp; Segata N (2021) Integrating taxonomic, functional,
and strain-level profiling of diverse microbial communities with bioBakery 3.
<em>eLife</em>. 10:e65088. doi: 10.7554/eLife.65088</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="importHUMAnN.html">importHUMAnN</a></code>
<code><a href="makeTreeSEFromPhyloseq.html">makeTreeSEFromPhyloseq</a></code>
<code><a href="makeTreeSEFromBiom.html">makeTreeSEFromBiom</a></code>
<code><a href="makeTreeSEFromDADA2.html">makeTreeSEFromDADA2</a></code>
<code><a href="importQIIME2.html">importQIIME2</a></code>
<code><a href="importMothur.html">importMothur</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Leo Lahti and Tuomas Borman. Contact: <a href="microbiome.github.io">microbiome.github.io</a></p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># (Data is from tutorial</span></span></span>
<span class="r-in"><span><span class="co"># https://github.com/biobakery/biobakery/wiki/metaphlan3#merge-outputs)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># File path</span></span></span>
<span class="r-in"><span><span class="va">file_path</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"merged_abundance_table.txt"</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Import data</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu">importMetaPhlAn</span><span class="op">(</span><span class="va">file_path</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># Data at the lowest rank</span></span></span>
<span class="r-in"><span><span class="va">tse</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 16 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(16): s__Corynebacterium_matruchotii s__Rothia_dentocariosa ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   s__Haemophilus_haemolyticus s__Moraxella_nonliquefaciens</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(9): Kingdom Phylum ... clade_name NCBI_tax_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(6): SRS014494-Posterior_fornix_profile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SRS014476-Supragingival_plaque_profile ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SRS014464-Anterior_nares_profile SRS014459-Stool_profile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(6): Kingdom Phylum ... Family Genus</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span><span class="co"># Data at higher rank is stored in altExp</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExps</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> List of length 6</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> names(6): Kingdom Phylum Class Order Family Genus</span>
<span class="r-in"><span><span class="co"># Higher rank data is in SE format, for example, Phylum rank</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"Phylum"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 4 6 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(1): counts</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(4): p__Actinobacteria p__Bacteroidetes p__Firmicutes</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   p__Proteobacteria</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(9): Kingdom Phylum ... clade_name NCBI_tax_id</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(6): SRS014494-Posterior_fornix_profile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SRS014476-Supragingival_plaque_profile ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SRS014464-Anterior_nares_profile SRS014459-Stool_profile</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer></div>

  


  

  </body></html>

