<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Subsample Counts — rarefyAssay • mia</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Subsample Counts — rarefyAssay"><meta property="og:description" content="rarefyAssay randomly subsamples counts within a
SummarizedExperiment object and returns a new
SummarizedExperiment containing the original assay and the new
subsampled assay."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">


    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.23</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Subsample Counts</h1>
    <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/master/R/AllGenerics.R" class="external-link"><code>R/AllGenerics.R</code></a>, <a href="https://github.com/microbiome/mia/blob/master/R/rarefyAssay.R" class="external-link"><code>R/rarefyAssay.R</code></a></small>
    <div class="hidden name"><code>rarefyAssay.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><code>rarefyAssay</code> randomly subsamples counts within a
<code>SummarizedExperiment</code> object and returns a new
<code>SummarizedExperiment</code> containing the original assay and the new
subsampled assay.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">rarefyAssay</span><span class="op">(</span><span class="va">x</span>, <span class="va">...</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># S4 method for class 'SummarizedExperiment'</span></span>
<span><span class="fu">rarefyAssay</span><span class="op">(</span></span>
<span>  <span class="va">x</span>,</span>
<span>  assay.type <span class="op">=</span> <span class="va">assay_name</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  sample <span class="op">=</span> <span class="va">min_size</span>,</span>
<span>  min_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fu">colSums2</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">x</span>, <span class="va">assay.type</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>,</span>
<span>  replace <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  name <span class="op">=</span> <span class="st">"subsampled"</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>


<dl><dt id="arg-x">x<a class="anchor" aria-label="anchor" href="#arg-x"></a></dt>
<dd><p><code><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-class.html" class="external-link">TreeSummarizedExperiment</a></code>.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>optional arguments:</p><ul><li><p><code>verbose</code>: <code>Logical scalar</code>. Choose whether to show
messages. (Default: <code>TRUE</code>)</p></li>
</ul></dd>


<dt id="arg-assay-type">assay.type<a class="anchor" aria-label="anchor" href="#arg-assay-type"></a></dt>
<dd><p><code>Character scalar</code>. Specifies the name of assay
used in calculation. (Default: <code>"counts"</code>)</p></dd>


<dt id="arg-assay-name">assay_name<a class="anchor" aria-label="anchor" href="#arg-assay-name"></a></dt>
<dd><p>Deprecated. Use <code>assay.type</code> instead.</p></dd>


<dt id="arg-sample">sample<a class="anchor" aria-label="anchor" href="#arg-sample"></a></dt>
<dd><p><code>Integer scalar</code>. Indicates the number of counts being
simulated i.e. rarefying depth. This can equal to lowest number of total
counts found in a sample or a user specified number.</p></dd>


<dt id="arg-min-size">min_size<a class="anchor" aria-label="anchor" href="#arg-min-size"></a></dt>
<dd><p>Deprecated. Use <code>sample</code> instead.</p></dd>


<dt id="arg-replace">replace<a class="anchor" aria-label="anchor" href="#arg-replace"></a></dt>
<dd><p><code>Logical scalar</code>. Whether to åperform subsampling with
replacement. Ths works similarly to <code>sample(..., replace = TRUE)</code>.
(Default: <code>FALSE</code>)</p></dd>


<dt id="arg-name">name<a class="anchor" aria-label="anchor" href="#arg-name"></a></dt>
<dd><p><code>Character scalar</code>. The name for the transformed assay to
be stored. (Default: <code>method</code>)</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p><code>rarefyAssay</code> return <code>x</code> with subsampled data.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Although the subsampling approach is highly debated in microbiome research,
we include the <code>rarefyAssay</code> function because there may be some
instances where it can be useful.
Note that the output of <code>rarefyAssay</code> is not the equivalent as the
input and any result have to be verified with the original dataset.</p>
<p>Subsampling/Rarefying may undermine downstream analyses and have unintended
consequences. Therefore, make sure this normalization is appropriate for
your data.</p>
<p>To maintain the reproducibility, please define the seed using set.seed()
before implement this function.</p>
<p>When <code>replace = FALSE</code>, the function uses internally
<code><a href="https://rdrr.io/pkg/vegan/man/rarefy.html" class="external-link">vegan::rarefy</a></code> while with replacement enabled the function utilizes
own implementation, inspired by <code><a href="https://rdrr.io/pkg/phyloseq/man/rarefy_even_depth.html" class="external-link">phyloseq::rarefy_even_depth</a></code>.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>McMurdie PJ, Holmes S. Waste not, want not: why rarefying microbiome data
is inadmissible. PLoS computational biology. 2014 Apr 3;10(4):e1003531.</p>
<p>Gloor GB, Macklaim JM, Pawlowsky-Glahn V &amp; Egozcue JJ (2017)
Microbiome Datasets Are Compositional: And This Is Not Optional.
Frontiers in Microbiology 8: 2224. doi: 10.3389/fmicb.2017.02224</p>
<p>Weiss S, Xu ZZ, Peddada S, Amir A, Bittinger K, Gonzalez A, Lozupone C,
Zaneveld JR, Vázquez-Baeza Y, Birmingham A, Hyde ER. Normalization and
microbial differential abundance strategies depend upon data characteristics.
Microbiome. 2017 Dec;5(1):1-8.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index">
<ul><li><p><code><a href="https://rdrr.io/pkg/vegan/man/rarefy.html" class="external-link">vegan::rrarefy</a></code></p></li>
<li><p><code><a href="https://rdrr.io/pkg/phyloseq/man/rarefy_even_depth.html" class="external-link">phyloseq::rarefy_even_depth</a></code></p></li>
</ul></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># When samples in TreeSE are less than specified sample, they will be</span></span></span>
<span class="r-in"><span><span class="co"># removed. If after subsampling features are not present in any of the</span></span></span>
<span class="r-in"><span><span class="co"># samples, they will be removed.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">tse_subsampled</span> <span class="op">&lt;-</span> <span class="fu">rarefyAssay</span><span class="op">(</span><span class="va">tse</span>, sample <span class="op">=</span> <span class="fl">60000</span>, name <span class="op">=</span> <span class="st">"subsampled"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1 samples removed because they contained fewer reads than `sample`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 6726 features removed because they are not present in any of the samples after subsampling.</span>
<span class="r-in"><span><span class="va">tse_subsampled</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> class: TreeSummarizedExperiment </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> dim: 12490 25 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> metadata(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> assays(2): counts subsampled</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rownames(12490): 549322 522457 ... 200359 271582</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowData names(7): Kingdom Phylum ... Genus Species</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colnames(25): CL3 CC1 ... Even2 Even3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colData names(7): X.SampleID Primer ... SampleType Description</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> reducedDimNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> mainExpName: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> altExpNames(0):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowLinks: a LinkDataFrame (12490 rows)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> rowTree: 1 phylo tree(s) (19216 leaves)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colLinks: NULL</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> colTree: NULL</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 19216    26</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">tse_subsampled</span>, <span class="st">"subsampled"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 12490    25</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Tuomas Borman, Felix G.M. Ernst, Sudarshan A. Shetty, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer></div>






  </body></html>

