<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>mia: Microbiome analysis tools • mia</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="mia: Microbiome analysis tools">
<meta property="og:description" content="mia">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">mia</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.13.22</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/mia.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/microbiome/mia/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>mia: Microbiome analysis tools</h1>
            
            <h4 data-toc-skip class="date">2024-06-17</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/microbiome/mia/blob/HEAD/vignettes/mia.Rmd" class="external-link"><code>vignettes/mia.Rmd</code></a></small>
      <div class="hidden name"><code>mia.Rmd</code></div>

    </div>

    
    
<p><code>mia</code> implements tools for microbiome analysis based on
the <code>SummarizedExperiment</code> <span class="citation">(Morgan et
al. 2020)</span>, <code>SingleCellExperiment</code> <span class="citation">(Amezquita et al. 2020)</span> and
<code>TreeSummarizedExperiment</code> <span class="citation">(Huang
2021)</span> infrastructure. Data wrangling and analysis are the main
scope of this package.</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install <code>mia</code>, install <code>BiocManager</code> first,
if it is not installed. Afterwards use the <code>install</code> function
from <code>BiocManager</code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"mia"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="load-mia">Load <em>mia</em><a class="anchor" aria-label="anchor" href="#load-mia"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/microbiome/mia" class="external-link">"mia"</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-a-treesummarizedexperiment-object">Loading a <code>TreeSummarizedExperiment</code> object<a class="anchor" aria-label="anchor" href="#loading-a-treesummarizedexperiment-object"></a>
</h2>
<p>A few example datasets are available via <code>mia</code>. For this
vignette the <code>GlobalPatterns</code> dataset is loaded first.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="va">GlobalPatterns</span></span>
<span><span class="va">tse</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 19216 26 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(19216): 549322 522457 ... 200359 271582</span></span>
<span><span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">## colnames(26): CL3 CC1 ... Even2 Even3</span></span>
<span><span class="co">## colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (19216 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="functions-for-working-with-microbiome-data">Functions for working with microbiome data<a class="anchor" aria-label="anchor" href="#functions-for-working-with-microbiome-data"></a>
</h2>
<p>One of the main topics for analysing microbiome data is the
application of taxonomic data to describe features measured. The
interest lies in the connection between individual bacterial species and
their relation to each other.</p>
<p><code>mia</code> does not rely on a specific object type to hold
taxonomic data, but uses specific columns in the <code>rowData</code> of
a <code>TreeSummarizedExperiment</code> object.
<code>taxonomyRanks</code> can be used to construct a
<code>character</code> vector of available taxonomic levels. This can be
used, for example, for subsetting.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># print the available taxonomic ranks</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</span></span></code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Kingdom" "Phylum"  "Class"   "Order"   "Family"  "Genus"   "Species"</span></span></code></pre>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># subset to taxonomic data only</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="../reference/taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## DataFrame with 19216 rows and 7 columns</span></span>
<span><span class="co">##            Kingdom        Phylum        Class        Order        Family</span></span>
<span><span class="co">##        &lt;character&gt;   &lt;character&gt;  &lt;character&gt;  &lt;character&gt;   &lt;character&gt;</span></span>
<span><span class="co">## 549322     Archaea Crenarchaeota Thermoprotei           NA            NA</span></span>
<span><span class="co">## 522457     Archaea Crenarchaeota Thermoprotei           NA            NA</span></span>
<span><span class="co">## 951        Archaea Crenarchaeota Thermoprotei Sulfolobales Sulfolobaceae</span></span>
<span><span class="co">## 244423     Archaea Crenarchaeota        Sd-NA           NA            NA</span></span>
<span><span class="co">## 586076     Archaea Crenarchaeota        Sd-NA           NA            NA</span></span>
<span><span class="co">## ...            ...           ...          ...          ...           ...</span></span>
<span><span class="co">## 278222    Bacteria           SR1           NA           NA            NA</span></span>
<span><span class="co">## 463590    Bacteria           SR1           NA           NA            NA</span></span>
<span><span class="co">## 535321    Bacteria           SR1           NA           NA            NA</span></span>
<span><span class="co">## 200359    Bacteria           SR1           NA           NA            NA</span></span>
<span><span class="co">## 271582    Bacteria           SR1           NA           NA            NA</span></span>
<span><span class="co">##              Genus                Species</span></span>
<span><span class="co">##        &lt;character&gt;            &lt;character&gt;</span></span>
<span><span class="co">## 549322          NA                     NA</span></span>
<span><span class="co">## 522457          NA                     NA</span></span>
<span><span class="co">## 951     Sulfolobus Sulfolobusacidocalda..</span></span>
<span><span class="co">## 244423          NA                     NA</span></span>
<span><span class="co">## 586076          NA                     NA</span></span>
<span><span class="co">## ...            ...                    ...</span></span>
<span><span class="co">## 278222          NA                     NA</span></span>
<span><span class="co">## 463590          NA                     NA</span></span>
<span><span class="co">## 535321          NA                     NA</span></span>
<span><span class="co">## 200359          NA                     NA</span></span>
<span><span class="co">## 271582          NA                     NA</span></span></code></pre>
<p>The columns are recognized case insensitive. Additional functions are
available to check for validity of taxonomic information or generate
labels based on the taxonomic information.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="fu"><a href="../reference/taxonomy-methods.html">taxonomyRankEmpty</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"Species"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## FALSE  TRUE </span></span>
<span><span class="co">##  1413 17803</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu"><a href="../reference/taxonomy-methods.html">getTaxonomyLabels</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Class:Thermoprotei"               "Class:Thermoprotei_1"            </span></span>
<span><span class="co">## [3] "Species:Sulfolobusacidocaldarius" "Class:Sd-NA"                     </span></span>
<span><span class="co">## [5] "Class:Sd-NA_1"                    "Class:Sd-NA_2"</span></span></code></pre>
<p>For more details see the man page <code><a href="../reference/taxonomy-methods.html">?taxonomyRanks</a></code>.</p>
<div class="section level3">
<h3 id="merging-and-agglomeration-based-on-taxonomic-information-">Merging and agglomeration based on taxonomic information.<a class="anchor" aria-label="anchor" href="#merging-and-agglomeration-based-on-taxonomic-information-"></a>
</h3>
<p>Agglomeration of data based on these taxonomic descriptors can be
performed using functions implemented in <code>mia</code>. In addition
to the <code>aggValue</code> functions provide by
<code>TreeSummarizedExperiment</code> <code>mergeFeaturesByRank</code>
is available. <code>mergeFeaturesByRank</code> does not require tree
data to be present.</p>
<p><code>mergeFeaturesByRank</code> constructs a <code>factor</code> to
guide merging from the available taxonomic information. For more
information on merging have a look at the man page via
<code><a href="../reference/deprecate.html">?mergeFeatures</a></code>.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># agglomerate at the Family taxonomic rank</span></span>
<span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deprecate.html">mergeFeaturesByRank</a></span><span class="op">(</span><span class="va">tse</span>, rank <span class="op">=</span> <span class="st">"Family"</span><span class="op">)</span></span>
<span><span class="co">## How many taxa before/after agglomeration?</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 19216</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 341</span></span></code></pre>
<p>Tree data can also be shrunk alongside agglomeration, but this is
turned of by default.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># with agglomeration of the tree</span></span>
<span><span class="va">x2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deprecate.html">mergeFeaturesByRank</a></span><span class="op">(</span><span class="va">tse</span>, rank <span class="op">=</span> <span class="st">"Family"</span>,</span>
<span>                        agglomerateTree <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span> <span class="co"># same number of rows, but</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 341</span></span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html" class="external-link">rowTree</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span> <span class="co"># ... different</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Phylogenetic tree with 19216 tips and 19215 internal nodes.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Tip labels:</span></span>
<span><span class="co">##   549322, 522457, 951, 244423, 586076, 246140, ...</span></span>
<span><span class="co">## Node labels:</span></span>
<span><span class="co">##   , 0.858.4, 1.000.154, 0.764.3, 0.995.2, 1.000.2, ...</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Rooted; includes branch lengths.</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/TreeSummarizedExperiment-accessor.html" class="external-link">rowTree</a></span><span class="op">(</span><span class="va">x2</span><span class="op">)</span> <span class="co"># ... tree</span></span></code></pre></div>
<pre><code><span><span class="co">## </span></span>
<span><span class="co">## Phylogenetic tree with 341 tips and 340 internal nodes.</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Tip labels:</span></span>
<span><span class="co">##   951, 215972, 138353, 546313, 173903, 202347, ...</span></span>
<span><span class="co">## Node labels:</span></span>
<span><span class="co">##   , 0.858.4, 0.764.3, 0.985.6, 1.000.112, 0.978.18, ...</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Rooted; includes branch lengths.</span></span></code></pre>
<p>For <code>mergeFeaturesByRank</code> to work, taxonomic data must be
present. Even though only one rank is available for the
<code>enterotype</code> dataset, agglomeration can be performed
effectively de-duplicating entries for the genus level.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">enterotype</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Genus"</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/deprecate.html">mergeFeaturesByRank</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 551 280 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(551): Prosthecochloris Chloroflexus ... Syntrophococcus</span></span>
<span><span class="co">##   Mogibacterium</span></span>
<span><span class="co">## rowData names(1): Genus</span></span>
<span><span class="co">## colnames(280): AM.AD.1 AM.AD.2 ... TS98_V2 TS99.2_V2</span></span>
<span><span class="co">## colData names(9): Enterotype Sample_ID ... Age ClinicalStatus</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: NULL</span></span>
<span><span class="co">## rowTree: NULL</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
<p>To keep data tidy, the agglomerated data can be stored as an
alternative experiment in the object of origin. With this synchronized
sample subsetting becomes very easy.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>, <span class="st">"family"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">x2</span></span></code></pre></div>
<p>Keep in mind, that if you set <code>na.rm = TRUE</code>, rows with
<code>NA</code> or similar value (defined via the
<code>empty.fields</code> argument) will be removed. Depending on these
settings different number of rows will be returned.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deprecate.html">mergeFeaturesByRank</a></span><span class="op">(</span><span class="va">tse</span>, rank <span class="op">=</span> <span class="st">"Species"</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"species"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deprecate.html">mergeFeaturesByRank</a></span><span class="op">(</span><span class="va">tse</span>, rank <span class="op">=</span> <span class="st">"Species"</span>, na.rm <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">x1</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 944  26</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"species"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 2307   26</span></span></code></pre>
<p>For convenience the function <code>agglomerateByRanks</code> is
available, which agglomerates data on all <code>ranks</code> selected.
By default all available ranks will be used. The output is compatible to
be stored as alternative experiments.</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/agglomerate-methods.html">agglomerateByRanks</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="va">tse</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 19216 26 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(19216): 549322 522457 ... 200359 271582</span></span>
<span><span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">## colnames(26): CL3 CC1 ... Even2 Even3</span></span>
<span><span class="co">## colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(9): family species ... Genus Species</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (19216 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExpNames</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "family"  "species" "Kingdom" "Phylum"  "Class"   "Order"   "Family" </span></span>
<span><span class="co">## [8] "Genus"   "Species"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="constructing-a-tree-from-taxonomic-data">Constructing a tree from taxonomic data<a class="anchor" aria-label="anchor" href="#constructing-a-tree-from-taxonomic-data"></a>
</h3>
<p>Constructing a taxonomic tree from taxonomic data stored in
<code>rowData</code> is quite straightforward and uses mostly functions
implemented in <code>TreeSummarizedExperiment</code>.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Species"</span><span class="op">)</span><span class="op">)</span><span class="op">[</span>,<span class="fu"><a href="../reference/taxonomy-methods.html">taxonomyRanks</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">taxa_res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/resolveLoop.html" class="external-link">resolveLoop</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">taxa</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">taxa_tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/toTree.html" class="external-link">toTree</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">taxa_res</span><span class="op">)</span></span>
<span><span class="va">taxa_tree</span><span class="op">$</span><span class="va">tip.label</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/taxonomy-methods.html">getTaxonomyLabels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Species"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">rowNodeLab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/taxonomy-methods.html">getTaxonomyLabels</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Species"</span><span class="op">)</span>, make_unique <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Species"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/TreeSummarizedExperiment/man/changeTree.html" class="external-link">changeTree</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Species"</span><span class="op">)</span>,</span>
<span>                                   rowTree <span class="op">=</span> <span class="va">taxa_tree</span>,</span>
<span>                                   rowNodeLab <span class="op">=</span> <span class="va">rowNodeLab</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="transformation-of-assay-data">Transformation of assay data<a class="anchor" aria-label="anchor" href="#transformation-of-assay-data"></a>
</h3>
<p>Transformation of count data stored in <code>assays</code> is also a
main task when work with microbiome data. <code>transformAssay</code>
can be used for this and offers a few choices of available
transformations. A modified object is returned and the transformed
counts are stored in a new <code>assay</code>.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "counts"</span></span></code></pre>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">anterotype</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/transformAssay.html">transformAssay</a></span><span class="op">(</span><span class="va">enterotype</span>, method <span class="op">=</span> <span class="st">"log10"</span>, pseudocount <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assayNames</a></span><span class="op">(</span><span class="va">enterotype</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "counts"</span></span></code></pre>
<p>For more details have a look at the man page
<code><a href="../reference/transformAssay.html">?transformAssay</a></code>.</p>
<p>Sub-sampling to equal number of counts per sample. Also known as
rarefying.</p>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tse.subsampled</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/rarefyAssay.html">rarefyAssay</a></span><span class="op">(</span><span class="va">GlobalPatterns</span>, </span>
<span>                                  min_size <span class="op">=</span> <span class="fl">60000</span>, </span>
<span>                                  name <span class="op">=</span> <span class="st">"subsampled"</span>,</span>
<span>                                  replace <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  seed <span class="op">=</span> <span class="fl">1938</span><span class="op">)</span></span>
<span><span class="va">tse.subsampled</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 12474 25 </span></span>
<span><span class="co">## metadata(1): subsampleCounts_min_size</span></span>
<span><span class="co">## assays(2): counts subsampled</span></span>
<span><span class="co">## rownames(12474): 549322 244423 ... 200359 271582</span></span>
<span><span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">## colnames(25): CL3 CC1 ... Even2 Even3</span></span>
<span><span class="co">## colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (12474 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
<p>Alternatively, one can save both original TreeSE and subsampled
TreeSE within a MultiAssayExperiment object.</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://waldronlab.io/MultiAssayExperiment/" class="external-link">MultiAssayExperiment</a></span><span class="op">)</span></span>
<span><span class="va">mae</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment.html" class="external-link">MultiAssayExperiment</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"originalTreeSE"</span> <span class="op">=</span> <span class="va">GlobalPatterns</span>,</span>
<span>                              <span class="st">"subsampledTreeSE"</span> <span class="op">=</span> <span class="va">tse.subsampled</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mae</span></span></code></pre></div>
<pre><code><span><span class="co">## A MultiAssayExperiment object of 2 listed</span></span>
<span><span class="co">##  experiments with user-defined names and respective classes.</span></span>
<span><span class="co">##  Containing an ExperimentList class object of length 2:</span></span>
<span><span class="co">##  [1] originalTreeSE: TreeSummarizedExperiment with 19216 rows and 26 columns</span></span>
<span><span class="co">##  [2] subsampledTreeSE: TreeSummarizedExperiment with 12474 rows and 25 columns</span></span>
<span><span class="co">## Functionality:</span></span>
<span><span class="co">##  experiments() - obtain the ExperimentList instance</span></span>
<span><span class="co">##  colData() - the primary/phenotype DataFrame</span></span>
<span><span class="co">##  sampleMap() - the sample coordination DataFrame</span></span>
<span><span class="co">##  `$`, `[`, `[[` - extract colData columns, subset, or experiment</span></span>
<span><span class="co">##  *Format() - convert into a long or wide DataFrame</span></span>
<span><span class="co">##  assays() - convert ExperimentList to a SimpleList of matrices</span></span>
<span><span class="co">##  exportClass() - save data to flat files</span></span></code></pre>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># To extract specifically the subsampled TreeSE</span></span>
<span><span class="fu"><a href="http://waldronlab.io/MultiAssayExperiment/reference/MultiAssayExperiment-methods.html" class="external-link">experiments</a></span><span class="op">(</span><span class="va">mae</span><span class="op">)</span><span class="op">$</span><span class="va">subsampledTreeSE</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 12474 25 </span></span>
<span><span class="co">## metadata(1): subsampleCounts_min_size</span></span>
<span><span class="co">## assays(2): counts subsampled</span></span>
<span><span class="co">## rownames(12474): 549322 244423 ... 200359 271582</span></span>
<span><span class="co">## rowData names(7): Kingdom Phylum ... Genus Species</span></span>
<span><span class="co">## colnames(25): CL3 CC1 ... Even2 Even3</span></span>
<span><span class="co">## colData names(7): X.SampleID Primer ... SampleType Description</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (12474 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (19216 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="community-indices">Community indices<a class="anchor" aria-label="anchor" href="#community-indices"></a>
</h3>
<p>In the field of microbiome ecology several indices to describe
samples and community of samples are available. In this vignette we just
want to give a very brief introduction.</p>
<p>Functions for calculating alpha and beta diversity indices are
available. Using <code>estimateDiversity</code> multiple diversity
indices are calculated by default and results are stored automatically
in <code>colData</code>. Selected indices can be calculated individually
by setting <code>index = "shannon"</code> for example.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/estimateDiversity.html">estimateDiversity</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">8</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">tse</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "coverage"            "faith"               "fisher"             </span></span>
<span><span class="co">## [4] "gini_simpson"        "inverse_simpson"     "log_modulo_skewness"</span></span>
<span><span class="co">## [7] "shannon"</span></span></code></pre>
<p>Beta diversity indices are used to describe inter-sample connections.
Technically they are calculated as <code>dist</code> object and reduced
dimensions can be extracted using <code>cmdscale</code>. This is wrapped
up in the <code>runMDS</code> function of the <code>scater</code>
package, but can be easily used to calculated beta diversity indices
using the established functions from the <code>vegan</code> package or
any other package using comparable inputs.</p>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/scater/" class="external-link">scater</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Genus"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/scater/man/runMDS.html" class="external-link">runMDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SingleCellExperiment/man/altExps.html" class="external-link">altExp</a></span><span class="op">(</span><span class="va">tse</span>,<span class="st">"Genus"</span><span class="op">)</span>, </span>
<span>                              FUN <span class="op">=</span> <span class="fu">vegan</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/pkg/vegan/man/vegdist.html" class="external-link">vegdist</a></span>,</span>
<span>                              method <span class="op">=</span> <span class="st">"bray"</span>,</span>
<span>                              name <span class="op">=</span> <span class="st">"BrayCurtis"</span>, </span>
<span>                              ncomponents <span class="op">=</span> <span class="fl">5</span>, </span>
<span>                              assay.type <span class="op">=</span> <span class="st">"counts"</span>, </span>
<span>                              keep_dist <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>JSD and UniFrac are implemented in <code>mia</code> as well.
<code>calculateJSD</code> can be used as a drop-in replacement in the
example above (omit the <code>method</code> argument as well) to
calculate the JSD. For calculating the UniFrac distance via
<code>calculateUniFrac</code> either a
<code>TreeSummarizedExperiment</code> must be used or a tree supplied
via the <code>tree</code> argument. For more details see
<code><a href="../reference/calculateJSD.html">?calculateJSD</a></code>, <code><a href="../reference/calculateUnifrac.html">?calculateUnifrac</a></code> or
<code><a href="../reference/runDPCoA.html">?getDPCoA</a></code>.</p>
<p><code>runMDS</code> performs the decomposition. Alternatively
<code>addNMDS</code> can also be used.</p>
</div>
<div class="section level3">
<h3 id="other-indices">Other indices<a class="anchor" aria-label="anchor" href="#other-indices"></a>
</h3>
<p><code>estimateDominance</code> and <code>estimateEvenness</code>
implement other sample-wise indices. The function behave equivalently to
<code>estimateDiversity</code>. For more information see the
corresponding man pages.</p>
</div>
</div>
<div class="section level2">
<h2 id="utility-functions">Utility functions<a class="anchor" aria-label="anchor" href="#utility-functions"></a>
</h2>
<p>To make migration and adoption as easy as possible several utility
functions are available.</p>
<div class="section level3">
<h3 id="data-loading-functions">Data loading functions<a class="anchor" aria-label="anchor" href="#data-loading-functions"></a>
</h3>
<p>Functions to load data from <code>biom</code> files,
<code>QIIME2</code> output, <code>DADA2</code> objects <span class="citation">(Callahan et al. 2016)</span> or <code>phyloseq</code>
objects are available.</p>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217" class="external-link">phyloseq</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">esophagus</span>, package <span class="op">=</span> <span class="st">"phyloseq"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">esophagus</span></span></code></pre></div>
<pre><code><span><span class="co">## phyloseq-class experiment-level object</span></span>
<span><span class="co">## otu_table()   OTU Table:         [ 58 taxa and 3 samples ]</span></span>
<span><span class="co">## phy_tree()    Phylogenetic Tree: [ 58 tips and 57 internal nodes ]</span></span></code></pre>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">esophagus</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/makeTreeSEFromPhyloseq.html">makeTreeSEFromPhyloseq</a></span><span class="op">(</span><span class="va">esophagus</span><span class="op">)</span></span>
<span><span class="va">esophagus</span></span></code></pre></div>
<pre><code><span><span class="co">## class: TreeSummarizedExperiment </span></span>
<span><span class="co">## dim: 58 3 </span></span>
<span><span class="co">## metadata(0):</span></span>
<span><span class="co">## assays(1): counts</span></span>
<span><span class="co">## rownames(58): 59_8_22 59_5_13 ... 65_9_9 59_2_6</span></span>
<span><span class="co">## rowData names(0):</span></span>
<span><span class="co">## colnames(3): B C D</span></span>
<span><span class="co">## colData names(0):</span></span>
<span><span class="co">## reducedDimNames(0):</span></span>
<span><span class="co">## mainExpName: NULL</span></span>
<span><span class="co">## altExpNames(0):</span></span>
<span><span class="co">## rowLinks: a LinkDataFrame (58 rows)</span></span>
<span><span class="co">## rowTree: 1 phylo tree(s) (58 leaves)</span></span>
<span><span class="co">## colLinks: NULL</span></span>
<span><span class="co">## colTree: NULL</span></span></code></pre>
<p>For more details have a look at the man page, for examples
<code><a href="../reference/makeTreeSEFromPhyloseq.html">?makeTreeSEFromPhyloseq</a></code>.</p>
</div>
<div class="section level3">
<h3 id="general-wrapper-functions">General wrapper functions<a class="anchor" aria-label="anchor" href="#general-wrapper-functions"></a>
</h3>
<p>Row-wise or column-wise assay data subsetting.</p>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Specific taxa</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">tse</span><span class="op">[</span><span class="st">'522457'</span>,<span class="op">]</span>, <span class="st">"counts"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        CL3 CC1 SV1 M31Fcsw M11Fcsw M31Plmr M11Plmr F21Plmr M31Tong M11Tong</span></span>
<span><span class="co">## 522457   0   0   0       0       0       0       0       0       0       0</span></span>
<span><span class="co">##        LMEpi24M SLEpi20M AQC1cm AQC4cm AQC7cm NP2 NP3 NP5 TRRsed1 TRRsed2</span></span>
<span><span class="co">## 522457        0        0      0      2      6   0   0   0       0       0</span></span>
<span><span class="co">##        TRRsed3 TS28 TS29 Even1 Even2 Even3</span></span>
<span><span class="co">## 522457       0    0    0     0     0     0</span></span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Specific sample</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">tse</span><span class="op">[</span>,<span class="st">'CC1'</span><span class="op">]</span>, <span class="st">"counts"</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##        CC1</span></span>
<span><span class="co">## 549322   0</span></span>
<span><span class="co">## 522457   0</span></span>
<span><span class="co">## 951      0</span></span>
<span><span class="co">## 244423   0</span></span>
<span><span class="co">## 586076   0</span></span>
<span><span class="co">## 246140   0</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="selecting-most-interesting-features">Selecting most interesting features<a class="anchor" aria-label="anchor" href="#selecting-most-interesting-features"></a>
</h3>
<p><code>getTop</code> returns a vector of the most <code>top</code>
abundant feature IDs.</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">esophagus</span>, package <span class="op">=</span> <span class="st">"mia"</span><span class="op">)</span></span>
<span><span class="va">top_taxa</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/summaries.html">getTop</a></span><span class="op">(</span><span class="va">esophagus</span>,</span>
<span>                    method <span class="op">=</span> <span class="st">"mean"</span>,</span>
<span>                    top <span class="op">=</span> <span class="fl">5</span>,</span>
<span>                    assay.type <span class="op">=</span> <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="va">top_taxa</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "59_2_6"  "59_7_6"  "59_8_22" "59_5_19" "65_6_2"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="generating-tidy-data">Generating tidy data<a class="anchor" aria-label="anchor" href="#generating-tidy-data"></a>
</h3>
<p>To generate tidy data as used and required in most of the tidyverse,
<code>meltAssay</code> can be used. A <code>data.frame</code> in the
long format will be returned.</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">molten_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deprecate.html">meltAssay</a></span><span class="op">(</span><span class="va">tse</span>,</span>
<span>                         assay.type <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>                         add_row_data <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                         add_col_data <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="va">molten_data</span></span></code></pre></div>
<pre><code><span><span class="co">## <span style="color: #949494;"># A tibble: 499,616 × 24</span></span></span>
<span><span class="co">##    FeatureID SampleID counts Kingdom Phylum     Class Order Family Genus Species</span></span>
<span><span class="co">##    <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 1</span> 549322    CL3           0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 2</span> 549322    CC1           0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 3</span> 549322    SV1           0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 4</span> 549322    M31Fcsw       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 5</span> 549322    M11Fcsw       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 6</span> 549322    M31Plmr       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 7</span> 549322    M11Plmr       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 8</span> 549322    F21Plmr       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;"> 9</span> 549322    M31Tong       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #BCBCBC;">10</span> 549322    M11Tong       0 Archaea Crenarcha… Ther… <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     <span style="color: #BB0000;">NA</span>    <span style="color: #BB0000;">NA</span>     </span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 499,606 more rows</span></span></span>
<span><span class="co">## <span style="color: #949494;"># ℹ 14 more variables: X.SampleID &lt;fct&gt;, Primer &lt;fct&gt;, Final_Barcode &lt;fct&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   Barcode_truncated_plus_T &lt;fct&gt;, Barcode_full_length &lt;fct&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   SampleType &lt;fct&gt;, Description &lt;fct&gt;, coverage &lt;int&gt;, faith &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   fisher &lt;dbl&gt;, gini_simpson &lt;dbl&gt;, inverse_simpson &lt;dbl&gt;,</span></span></span>
<span><span class="co">## <span style="color: #949494;">#   log_modulo_skewness &lt;dbl&gt;, shannon &lt;dbl&gt;</span></span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] phyloseq_1.49.0                 scater_1.33.1                  </span></span>
<span><span class="co">##  [3] ggplot2_3.5.1                   scuttle_1.15.0                 </span></span>
<span><span class="co">##  [5] mia_1.13.22                     MultiAssayExperiment_1.31.3    </span></span>
<span><span class="co">##  [7] TreeSummarizedExperiment_2.13.0 Biostrings_2.73.1              </span></span>
<span><span class="co">##  [9] XVector_0.45.0                  SingleCellExperiment_1.27.2    </span></span>
<span><span class="co">## [11] SummarizedExperiment_1.35.0     Biobase_2.65.0                 </span></span>
<span><span class="co">## [13] GenomicRanges_1.57.1            GenomeInfoDb_1.41.1            </span></span>
<span><span class="co">## [15] IRanges_2.39.0                  S4Vectors_0.43.0               </span></span>
<span><span class="co">## [17] BiocGenerics_0.51.0             MatrixGenerics_1.17.0          </span></span>
<span><span class="co">## [19] matrixStats_1.3.0               BiocStyle_2.33.1               </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rstudioapi_0.16.0           jsonlite_1.8.8             </span></span>
<span><span class="co">##   [3] magrittr_2.0.3              ggbeeswarm_0.7.2           </span></span>
<span><span class="co">##   [5] nloptr_2.0.3                rmarkdown_2.27             </span></span>
<span><span class="co">##   [7] fs_1.6.4                    zlibbioc_1.51.1            </span></span>
<span><span class="co">##   [9] ragg_1.3.2                  vctrs_0.6.5                </span></span>
<span><span class="co">##  [11] multtest_2.61.0             memoise_2.0.1              </span></span>
<span><span class="co">##  [13] minqa_1.2.7                 DelayedMatrixStats_1.27.1  </span></span>
<span><span class="co">##  [15] base64enc_0.1-3             htmltools_0.5.8.1          </span></span>
<span><span class="co">##  [17] S4Arrays_1.5.1              BiocNeighbors_1.23.0       </span></span>
<span><span class="co">##  [19] Rhdf5lib_1.27.0             rhdf5_2.49.0               </span></span>
<span><span class="co">##  [21] SparseArray_1.5.8           Formula_1.2-5              </span></span>
<span><span class="co">##  [23] sass_0.4.9                  bslib_0.7.0                </span></span>
<span><span class="co">##  [25] htmlwidgets_1.6.4           desc_1.4.3                 </span></span>
<span><span class="co">##  [27] sandwich_3.1-0              plyr_1.8.9                 </span></span>
<span><span class="co">##  [29] DECIPHER_3.1.4              zoo_1.8-12                 </span></span>
<span><span class="co">##  [31] cachem_1.1.0                igraph_2.0.3               </span></span>
<span><span class="co">##  [33] iterators_1.0.14            lifecycle_1.0.4            </span></span>
<span><span class="co">##  [35] pkgconfig_2.0.3             rsvd_1.0.5                 </span></span>
<span><span class="co">##  [37] Matrix_1.7-0                R6_2.5.1                   </span></span>
<span><span class="co">##  [39] fastmap_1.2.0               GenomeInfoDbData_1.2.12    </span></span>
<span><span class="co">##  [41] digest_0.6.35               colorspace_2.1-0           </span></span>
<span><span class="co">##  [43] irlba_2.3.5.1               textshaping_0.4.0          </span></span>
<span><span class="co">##  [45] Hmisc_5.1-3                 vegan_2.6-6.1              </span></span>
<span><span class="co">##  [47] beachmat_2.21.3             fansi_1.0.6                </span></span>
<span><span class="co">##  [49] mgcv_1.9-1                  httr_1.4.7                 </span></span>
<span><span class="co">##  [51] abind_1.4-5                 compiler_4.4.0             </span></span>
<span><span class="co">##  [53] withr_3.0.0                 htmlTable_2.4.2            </span></span>
<span><span class="co">##  [55] backports_1.5.0             BiocParallel_1.39.0        </span></span>
<span><span class="co">##  [57] viridis_0.6.5               DBI_1.2.3                  </span></span>
<span><span class="co">##  [59] MASS_7.3-60.2               DelayedArray_0.31.1        </span></span>
<span><span class="co">##  [61] biomformat_1.33.0           bluster_1.15.0             </span></span>
<span><span class="co">##  [63] permute_0.9-7               tools_4.4.0                </span></span>
<span><span class="co">##  [65] vipor_0.4.7                 foreign_0.8-86             </span></span>
<span><span class="co">##  [67] beeswarm_0.4.0              ape_5.8                    </span></span>
<span><span class="co">##  [69] nnet_7.3-19                 glue_1.7.0                 </span></span>
<span><span class="co">##  [71] rhdf5filters_1.17.0         nlme_3.1-165               </span></span>
<span><span class="co">##  [73] grid_4.4.0                  checkmate_2.3.1            </span></span>
<span><span class="co">##  [75] ade4_1.7-22                 cluster_2.1.6              </span></span>
<span><span class="co">##  [77] reshape2_1.4.4              lpSolve_5.6.20             </span></span>
<span><span class="co">##  [79] generics_0.1.3              gtable_0.3.5               </span></span>
<span><span class="co">##  [81] mediation_4.5.0             tidyr_1.3.1                </span></span>
<span><span class="co">##  [83] data.table_1.15.4           BiocSingular_1.21.1        </span></span>
<span><span class="co">##  [85] ScaledMatrix_1.13.0         utf8_1.2.4                 </span></span>
<span><span class="co">##  [87] foreach_1.5.2               ggrepel_0.9.5              </span></span>
<span><span class="co">##  [89] pillar_1.9.0                stringr_1.5.1              </span></span>
<span><span class="co">##  [91] yulab.utils_0.1.4           splines_4.4.0              </span></span>
<span><span class="co">##  [93] dplyr_1.1.4                 treeio_1.29.0              </span></span>
<span><span class="co">##  [95] lattice_0.22-6              survival_3.7-0             </span></span>
<span><span class="co">##  [97] tidyselect_1.2.1            DirichletMultinomial_1.47.0</span></span>
<span><span class="co">##  [99] knitr_1.47                  gridExtra_2.3              </span></span>
<span><span class="co">## [101] bookdown_0.39               xfun_0.44                  </span></span>
<span><span class="co">## [103] rbiom_1.0.3                 stringi_1.8.4              </span></span>
<span><span class="co">## [105] UCSC.utils_1.1.0            lazyeval_0.2.2             </span></span>
<span><span class="co">## [107] yaml_2.3.8                  boot_1.3-30                </span></span>
<span><span class="co">## [109] evaluate_0.24.0             codetools_0.2-20           </span></span>
<span><span class="co">## [111] tibble_3.2.1                BiocManager_1.30.23        </span></span>
<span><span class="co">## [113] cli_3.6.2                   RcppParallel_5.1.7         </span></span>
<span><span class="co">## [115] rpart_4.1.23                systemfonts_1.1.0          </span></span>
<span><span class="co">## [117] munsell_0.5.1               jquerylib_0.1.4            </span></span>
<span><span class="co">## [119] Rcpp_1.0.12                 parallel_4.4.0             </span></span>
<span><span class="co">## [121] pkgdown_2.0.9               sparseMatrixStats_1.17.2   </span></span>
<span><span class="co">## [123] lme4_1.1-35.3               slam_0.1-50                </span></span>
<span><span class="co">## [125] mvtnorm_1.2-5               decontam_1.25.0            </span></span>
<span><span class="co">## [127] viridisLite_0.4.2           tidytree_0.4.6             </span></span>
<span><span class="co">## [129] scales_1.3.0                purrr_1.0.2                </span></span>
<span><span class="co">## [131] crayon_1.5.2                rlang_1.1.4</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-SCE" class="csl-entry">
Amezquita, Robert, Aaron Lun, Etienne Becht, Vince Carey, Lindsay Carpp,
Ludwig Geistlinger, Federico Marini, et al. 2020. <span>“Orchestrating
Single-Cell Analysis with Bioconductor.”</span> <em>Nature Methods</em>
17: 137–45. <a href="https://www.nature.com/articles/s41592-019-0654-x" class="external-link">https://www.nature.com/articles/s41592-019-0654-x</a>.
</div>
<div id="ref-dada2" class="csl-entry">
Callahan, Benjamin J, Paul J McMurdie, Michael J Rosen, Andrew W Han,
Amy Jo A Johnson, and Susan P Holmes. 2016. <span>“DADA2:
High-Resolution Sample Inference from Illumina Amplicon Data.”</span>
<em>Nature Methods</em> 13: 581–83. <a href="https://doi.org/10.1038/nmeth.3869" class="external-link">https://doi.org/10.1038/nmeth.3869</a>.
</div>
<div id="ref-TSE" class="csl-entry">
Huang, Ruizhu. 2021. <em>TreeSummarizedExperiment:
TreeSummarizedExperiment: A S4 Class for Data with Tree Structures</em>.
</div>
<div id="ref-SE" class="csl-entry">
Morgan, Martin, Valerie Obenchain, Jim Hester, and Hervé Pagès. 2020.
<em>SummarizedExperiment: SummarizedExperiment Container</em>. <a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">https://bioconductor.org/packages/SummarizedExperiment</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Felix G.M. Ernst, Sudarshan A. Shetty, Tuomas Borman, Leo Lahti.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
